### PR TITLE
Palmas-TO spider reactor with improved date filtering

### PR DESCRIPTION
**Pull request by rennerocha**
It includes:
- Refactor of the code to make spider simpler and removing unneeded external libraries (like requests)
- Allow date filtering using `start_date` argument to aviod unnecessary requests

### CODE REVIEW

File: processing/data_collection/gazette/spiders/to_palmas.py [hunk 1]
```diff
@@ -1,75 +1,70 @@
...
+        for gazette in gazettes:
+            gazette_date = gazette.xpath("./td[2]/text()").get()
+            gazette_url = response.urljoin(gazette.css("a::attr(href)").get())
+
+            is_extra_edition = bool(gazette.xpath(".//*[contains(., 'Suplemento')]"))
+            if is_extra_edition:
+                parent_gazette_class_id = gazette.re_first(r"treegrid-parent-(\d+)")
+                parent_gazette_class = f"treegrid-{parent_gazette_class_id}"
+                main_edition_gazette = response.css(f".{parent_gazette_class}")
+                gazette_date = main_edition_gazette.xpath("./td[2]/text()").get()
```
Comment (giuliocc) [inline processing/data_collection/gazette/spiders/to_palmas.py:46]: What do you think of making this into a separate method like `date_from_previous_row`?

File: processing/data_collection/gazette/spiders/to_palmas.py [hunk 2]
```diff
@@ -1,75 +1,70 @@
...
-        li_list = response.css("div.diario-content-todos > ul > li")
-        for li in li_list:
-            edicao, data = li.xpath('.//*[@id="audio-titulo"]/text()').re(
-                r"(\d*)ª Edição de (.*$)"
+            item = Gazette(
+                date=dateparser.parse(gazette_date, languages=["pt"]).date(),
+                is_extra_edition=is_extra_edition,
+                territory_id=self.TERRITORY_ID,
+                scraped_at=datetime.datetime.utcnow(),
+                power="executive",
```
Comment (giuliocc) [inline processing/data_collection/gazette/spiders/to_palmas.py:53]: Checking around at some dates I found some examples of legislative data, ("PUBLICAÇÕES DA CÂMARA MUNICIPAL"). Please take look 27/02/2020 and 25/05/2020. I think the power should be `"executive_legislative"`.

**Review by giuliocc — CHANGES_REQUESTED** [review]
I ran the entire timespan without downloading files and the year of 2020 downloading them (because of disk space). No issues were found and the date filtering worked like a charm. Nice!

So, I'm just proposing some minor adjustments if you agree :)

File: processing/data_collection/gazette/spiders/to_palmas.py (hunk 2 above)
Comment (rennerocha) [reply processing/data_collection/gazette/spiders/to_palmas.py]: Good to know! I fixed that. Thanks.

File: processing/data_collection/gazette/spiders/to_palmas.py (hunk 1 above)
Comment (rennerocha) [reply processing/data_collection/gazette/spiders/to_palmas.py]: Date is not from the previous row, but from the parent edition. We can have more than one extra edition in one day. I splitted as a new method so it is clearer from where we are getting that date.

Comment (giuliocc) [reply processing/data_collection/gazette/spiders/to_palmas.py]: Didn't know that. Thanks :)

**Comment by giuliocc** [comment]
Looks very good to me. When I get to a computer I'll run it again to finish the review.

**Review by giuliocc — APPROVED** [review]

**Review by jvanz — APPROVED** [review]

**Pull request merged**